### PR TITLE
TPad: Always check if GetPainter() fCanvas and gVirtualX are not null

### PR DESCRIPTION
**Pull request by andriish**
Always check if GetPainter() fCanvas and gVirtualX are not nullptr before using them.

Should fix some crashes in TPad.

Tag @couet.

### CODE REVIEW

**Comment by lmoneta** [comment]
@phsft-bot build

**Comment by andriish** [comment]
Hi @couet ,

could you, please, have a look?
It seems the fails in chacks are just about the CI configuration.

Best regards,
Andrii

**Review by couet — APPROVED** [review]
It seems ok.

File: graf2d/gpad/src/TPad.cxx [hunk 1]
```diff
@@ -653,7 +653,7 @@ void TPad::Clear(Option_t *option)
       getchar();
    }
 
-   if (!gPad->IsBatch()) GetPainter()->ClearDrawable();
+   if (!gPad->IsBatch()&&GetPainter()) GetPainter()->ClearDrawable();
    if (gVirtualPS && gPad == gPad->GetCanvas()) gVirtualPS->NewPage();
```
Comment (couet) [inline graf2d/gpad/src/TPad.cxx:657]: These protections look fine. One little cosmetic thing. we usually put space around `&&` i.e.:
```
if (!gPad->IsBatch() && GetPainter()) ... 
```

Comment (andriish) [reply graf2d/gpad/src/TPad.cxx:657]: Fixed.

**Comment by couet** [comment]
@andriish can you update this one according to my comments ?

**Comment by andriish** [comment]
Hi @couet ,

Oh. Sorry. Spaces also BEFORE. Sorry.


Best regards,
Andrii

**Comment by couet** [comment]
Hi @andriish ,

Thanks for the space before :-) can you also look at the test structure I pointed ? 2 ifs like that on the same line makes it hard to read and not easy to set break point in gdb.
Best
Olivier

**Comment by andriish** [comment]
Hi @couet ,

OK. I don't see the comment, but I've fixed the only line that had two ifs.

Best regards,

Andrii

**Comment by couet** [comment]
That's correct thanks.

File: graf2d/gpad/src/TPad.cxx [hunk 2]
```diff
@@ -1057,7 +1057,7 @@ void TPad::CopyPixmap()
    int px, py;
    XYtoAbsPixel(fX1, fY2, px, py);
 
-   if (fPixmapID != -1)
+   if (fPixmapID != -1&& GetPainter())
```
Comment (couet) [inline graf2d/gpad/src/TPad.cxx:1060]: a space is missing here

**Comment by andriish** [comment]
Hi @couet ,

I've put spaces there and around couple other &&.

Andrii

**Comment by couet** [comment]
Great Thanks !

**Pull request merged**